test/boost/user_function_test.cc

```diff
@@ -24,6 +24,7 @@
 #include "test/lib/tmpdir.hh"
 #include "test/lib/exception_utils.hh"
 #include "test/lib/test_utils.hh"
+#include "boost/date_time/gregorian/gregorian_types.hpp"
```

Comment:
System includes should use angle brackets.

Patch can be more direct. It's not a dbuild problem. Instead of explaining where you hit the problem, explain what the problem is. This file uses boost::gregorian::bad_year but doesn't include it directly. Files should directly include their dependencies instead of inheriting them from random places, since those random places can change.